File: src/main/java/br/com/jiratorio/repository/LeadTimeConfigRepository.java
package br.com.jiratorio.repository;

import java.util.List;

import br.com.jiratorio.domain.LeadTimeConfig;
import java.util.Optional;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface LeadTimeConfigRepository extends CrudRepository<LeadTimeConfig, Long> {

    List<LeadTimeConfig> findByBoardId(Long boardId);

    Optional<LeadTimeConfig> findByIdAndBoardId(Long id, Long boardId);

}

File: src/main/kotlin/br/com/jiratorio/repository/LeadTimeConfigRepository.kt
package br.com.jiratorio.repository

import br.com.jiratorio.domain.entity.LeadTimeConfig
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface LeadTimeConfigRepository : CrudRepository<LeadTimeConfig, Long> {

    fun findByBoardId(boardId: Long): List<LeadTimeConfig>

    fun findByIdAndBoardId(id: Long, boardId: Long): LeadTimeConfig?

    @JvmDefault
    fun findByIdOrNull(id: Long): LeadTimeConfig? =
        findById(id).orElse(null)

}
